propagator/xray/lib/opentelemetry/propagator/aws_xray_trace.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  # This module is entended to only be used as an override for how to generate IDs to be compliant with XRay
+  module AWSXRayTrace
+    extend self
+
+    # Generates a valid trace identifier, a 16-byte string with at least one
+    # non-zero byte.
+    # AWS Docs: https://docs.aws.amazon.com/xray/latest/api/API_PutTraceSegments.html
+    # hi - 4 bytes timestamp, 4 bytes random (Mid)
+    # low - 8 bytes random.
+    # Since we include timestamp, impossible to be invalid.
+    # @return [bytes] a valid trace ID that is compliant with AWS XRay.
+    def generate_trace_id
+      time_hi = generate_time_byte
+      mid = generate_variable_byte_id(4)
+      low = generate_variable_byte_id(8)
```

Comment:
Could you replace this with:
```suggestion
      rest = generate_variable_byte_id(12)
```
Better yet would be:
```suggestion
      rest = RANDOM.bytes(12)
```
Since you don't need to avoid generating 0.

Reply:
Since I removed this function, i'm directly using RANDOM like suggested.